src/result/result.py

```diff
@@ -441,3 +441,31 @@ async def async_wrapper(*args: P.args, **kwargs: P.kwargs) -> Result[R, TBE]:
         return async_wrapper
 
     return decorator
+
+
+def is_ok(result: Result[T, E]) -> TypeGuard[Ok[T]]:
+    """A typeguard to check if a result is an Ok
+
+    Usage:
+    >>> result = Ok(1)
+    >>> err = Err("error")
+    >>> if is_ok(result):
+    >>>     reveal_type(result)  # Revealed type is 'Ok[builtins.int*]'
+    >>> elif is_err(err):
+    >>>     reveal_type(err)  # Revealed type is 'Err[builtins.str*]'
+    """
+    return result.is_ok()
+
+
+def is_err(result: Result[T, E]) -> TypeGuard[Err[E]]:
+    """A typeguard to check if a result is an Err
+
+    Usage:
+    >>> result = Ok(1)
+    >>> err = Err("error")
```

Comment:
Can we generalize this example to something like,
```
r: Result[int, str] = get_a_result()
if is_ok(r):
    r   # r is of type Ok[int]
elif is_err(r):
    r   # r is of type Err[str]
```

Just so we have a smaller example, with only 1 variable.

Reply:
applied your version